src/app/store/courses/courses.facade.ts

```diff
@@ -1,8 +1,59 @@
 import { Injectable } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { CoursesState } from '@app/store/courses/courses.reducer';
+import * as CoursesActions from '@app/store/courses/courses.actions';
+import {
+  getAllCourses,
+  getCourse,
+  getErrorMessage,
+  isAllCoursesLoadingSelector,
+  isSearchingStateSelector,
+  isSingleCourseLoadingSelector,
+} from '@app/store/courses/courses.selectors';
+import { Course } from '@shared/models/course.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CoursesStateFacade {
-  // Add your code here
+  public isAllCourseLoading$ = this.store.pipe(
+    select(isAllCoursesLoadingSelector)
+  );
+  public isSingleCourseLoading$ = this.store.pipe(
+    select(isSingleCourseLoadingSelector)
+  );
+  public isSearchingState$ = this.store.pipe(select(isSearchingStateSelector));
+  public allCourses$ = this.store.pipe(select(getAllCourses));
+  public course$ = this.store.pipe(select(getCourse));
+  public errorMessage$ = this.store.pipe(select(getErrorMessage));
```

Comment:
Are you sure this is used somewhere else?

Reply:
deleted!